test/ProjectService.Business.UnitTests/Commands/EditProjectCommandTests.cs

```diff
@@ -0,0 +1,202 @@
+                new List<Operation<EditProjectRequest>>
+                {
+                    new Operation<EditProjectRequest>(
+                        "replace",
+                        $"/{nameof(EditProjectRequest.DepartmentId)}",
+                        "",
+                        _departmentId)
+                }, new CamelCasePropertyNamesContractResolver());
+            _dbRequest = new JsonPatchDocument<DbProject>();
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            _response = new OperationResultResponse<bool>();
+            _response.Body = true;
+            _response.Status = OperationResultStatusType.FullSuccess;
+
+            _getDepartmentResponse = new Mock<IGetDepartmentResponse>();
+            _getDepartmentResponse.Setup(x => x.Id).Returns(_departmentId);
+            _getDepartmentResponse.Setup(x => x.Name).Returns("Name");
+
+            _operationResult = new Mock<IOperationResult<IGetDepartmentResponse>>();
+            _operationResult.Setup(x => x.Body).Returns(_getDepartmentResponse.Object);
+            _operationResult.Setup(x => x.IsSuccess).Returns(true);
+
+            _brokerResponse = new Mock<Response<IOperationResult<IGetDepartmentResponse>>>();
+            _brokerResponse.Setup(x => x.Message).Returns(_operationResult.Object);
+
+            _mocker = new AutoMocker();
+            _mocker
+                .Setup<IEditProjectValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(true);
+
+            _mocker
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin())
+                .Returns(true);
+
+            _mocker
+                .Setup<IAccessValidator, bool>(x => x.HasRights(It.IsAny<int>()))
+                .Returns(true);
+
+            _mocker
+                .Setup<IRequestClient<IGetDepartmentRequest>, Task<Response<IOperationResult<IGetDepartmentResponse>>>>(
+                x => x.GetResponse<IOperationResult<IGetDepartmentResponse>>(
+                    IGetDepartmentRequest.CreateObj(null, _departmentId), default, default))
+                .Returns(Task.FromResult(_brokerResponse.Object));
+
+            _mocker
+                .Setup<IPatchDbProjectMapper, JsonPatchDocument<DbProject>>(x =>
+                    x.Map(It.IsAny<JsonPatchDocument<EditProjectRequest>>()))
+                .Returns(_dbRequest);
+
+            _mocker
+                .Setup<IProjectRepository, bool>(x => x.Edit(It.IsAny<Guid>(), _dbRequest))
+                .Returns(true);
+
+            _command = _mocker.CreateInstance<EditProjectCommand>();
+        }
+
+        [Test]
+        public void SuccessCommandExecuteWhenRequesterHasRightsAndNotAdmin()
+        {
+            _mocker
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin())
+                .Returns(false);
+
+            SerializerAssert.AreEqual(_response, _command.Execute(It.IsAny<Guid>(), _request));
+        }
```

Comment:
why you didn't add the checking rights "_moker ....... x => x.HasRights"? in the name of your test you are checking the rights

Reply:
I set rights to admin and to another user in setUp by default. In the test, i rewrite mock data.